### PR TITLE
CI: Fix Pycodestyle warning W605

### PR DESCRIPTION
**Pull request by Naveenaidu**
W605 is a warning for invalid escape sequences.

In python 3.6, Unknown escapes consisting of `\` and an ASCII letter in regular expressions will
cause an warning.In the upcoming python versions this would be treated as an error.
Owing to it pycodestyle have added this check for future Python versions.

The alternative way to escape special characters now is to append a extra `\` infront of the
sequences. Escape the characters by double backticks.

Fixes https://github.com/coala/coala-bears/issues/2760

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Naveenaidu** [comment]
> commit message doesn't say why ignoring is appropriate solution

@jayvdb The reason I ignored the `W605`(Invalid escape sequence) was for a quick fix for travis CI. Which I realized was a wrong thing to do( Never silence an error to make your test pass :P)

I have been going through the past PR's from the pycodestyle repo and other repo's which were facing such issues. Looking at the history of out PR builds. Travis started to fail because of this error somewhere around December 2 -  2018. Before that W605 existed but never poised the current problem. I am yet to understand what caused it.

Reading the comment [here](https://github.com/PyCQA/pycodestyle/issues/766#issuecomment-451141810) and the [Issue](https://github.com/Python-Markdown/markdown/issues/611), the better solution to the current failure of Travis would be to either escape the sequences such as `'\d'` by adding an extra `\` or by `r'\d'`.

After trying out the above method on my fork, I realised that we have more than 4 files which needs to be fixed else the W605 warning will persist.

I would love to have your opinions on this.

**Comment by frextrite** [comment]
The best option right now would be to wrap all the invalid escape sequences with `r" "`, in all the 4 files. 

Moreover, why did PycodestyleBear start emitting W605 all of a sudden? I couldn't find any erroneous commit that could've resulted in this behavior. Did something change upstream?

**Comment by sladyn98** [comment]
According to me we could ignore the warning for now, if we run into regex expression issues later due to the wrapping of r'  ' we could fix it via the string literal append, as mentioned by @Naveenaidu.

**Comment by jayvdb** [comment]
Why hasnt this been updated??  At the very least your commit message is useless, and it is referencing an issue which was closed as a duplicate.  Stop looking for other people to guide you to the best solution, especially not from maintainers.  Find the best possible solution yourself, or listen to the best advice from other community members.

**Comment by Naveenaidu** [comment]
> Why hasnt this been updated?? At the very least your commit message is useless, and it is referencing an issue which was closed as a duplicate. Stop looking for other people to guide you to the best solution, especially not from maintainers. Find the best possible solution yourself, or listen to the best advice from other community members.

Thanks for telling me this :)

@sladyn98 IMHO Ignoring the warning didn't feel the best possible approach to the problem. I stand by my point that we should not ignore the errors only to make the build possible.

@frextrite  As discussed above, I first used a `r" "` to wrap all the invalid sequences in it. But then I faced two problems with the above solution
1. Sphinx documentation doesn't recognize `r" "`, and hence in the documentation, the sequences haven't been escaped rather they appeared as a string. eg:- (`r"/*"` appeared as `r"/*"`, whereas it should detect the expression as only `*` after the netlify build of documentation) (I would be raising an issue there to inform about it soon)
2. Pycodestyle still emitted the warning of `Invalid escape sequences`

That's the reason why i did not use `r` to wrap the invalid sequences.

An alternative solution to the above problem was to use an extra backslash `\` before the sequence. This solved the above two problems. Sphinx perfectly detected a `\\*` as `*` and Pycodestyle didn't complain too.

IMHO it is necessary to mention this somewhere in our documentation so that we don't encounter these errors in the future and cause build failures.

File: bears/c_languages/codeclone_detection/ClangCountingConditions.py [hunk 1]
```diff
@@ -245,17 +245,17 @@ def in_sum(stack):  # pragma nt: no cover
 def in_product(stack):  # pragma nt: no cover
     """
     A counting condition returning true if the variable is used in a product
-    statement, i.e. within the operators \*, /, % and their associated compound
-    operators.
+    statement, i.e. within the operators \\*, /, % and their associated
+    compound operators.
```
Comment (Naveenaidu) [inline bears/c_languages/codeclone_detection/ClangCountingConditions.py]: This change has been made to keep the character count under 80

Comment (Naveenaidu) [inline bears/c_languages/codeclone_detection/ClangCountingConditions.py]: This change has been made to keep the character count under 80. The same goes for the below change.

**Comment by sladyn98** [comment]
Yeah @Naveenaidu your right but  I suggested ignoring because you suggested that we would have to make changes in four files,else in my original comment on gitter I said that adding an extra '\' to avoid the escape sequences  error would be the best solution.Moreover @jayvdb mentioned in his orginal issue that sphinx would throw errors on ```r" " ``` wrapping ,my bad not to see that, anyways we would have to wait for the gometa linter issue to fixed for us to see the travis build and then proceed.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala-bears/pull/2848/commits/25a2c43f7f0785518134dadf098495d52795c316

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**